### ISSUE TITLE
Retail Confimation ID

### ISSUE THREAD

**Issue by dejl** [UMSKT/UMSKT#1]
The link in the description to the activation service is down.. is there a open source version of it?

**Comment by Neo-Desktop** [comment]
I don't think there's a website available but there is an executable that does work, I'm actually testing it as we speak and have been able to successfully generate a key for (and activate) windows xp pro sp3

The file you're looking for is named `xp_activate32.exe`

I'll direct you to this thread/comment
https://old.reddit.com/r/windowsxp/comments/wwjy5j/windows_xp_web_activation_is_finally_dead/ilmr5s3/

**Comment by Neo-Desktop** [comment]
I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version

**Comment by dejl** [comment]
> I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version

Installing XP in a VM right now and trying it!

**Comment by dejl** [comment]
> I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version

So I built your repo and ran ./xpkey on a Debian Linux system & all the keys generated say "product key is invalid", do I have to edit the main.cpp with a specific product ID or something for the edition I'm installing? in this case XP Pro?

**Comment by Neo-Desktop** [comment]
They product keys are for Windows XP Pro VL

Let me look into why compiling on debian is throwing that error.

I was testing the changes under WSL Ubuntu but that shouldn't be an issue

**Comment by dejl** [comment]
> They product keys are for Windows XP Pro VL
> 
> Let me look into why compiling on debian is throwing that error.
> 
> I was testing the changes under WSL Ubuntu but that shouldn't be an issue

I'm using a 64-bit Debian so, maybe you're using 32bit? Not sure.

**Comment by dejl** [comment]
> They product keys are for Windows XP Pro VL
> 
> Let me look into why compiling on debian is throwing that error.
> 
> I was testing the changes under WSL Ubuntu but that shouldn't be an issue

Oh to be clear the tool says the key is valid, but when I enter in XP install (or after install), it says invalid product key entered..

**Comment by TCB13** [comment]
> I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version

Thank you for this! Anyways, are the keys only usable in XP Pro VL? What about OEM versions?

**Comment by RetroSoftwareRepository** [comment]
> I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version

Ooh nice... also, I'm glad I could be of help for those who have retail XP discs laying around. I happen to have a XP SP2 system builder CD that I know (without the aid of this tool) would have been unusable today because activation actually doesn't work without updating certificates in the OS itself.

**Comment by Neo-Desktop** [comment]
@TCB13 
> > I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version
> 
> Thank you for this! Anyways, are the keys only usable in XP Pro VL? What about OEM versions?

Yes so I just test this on an XP Pro Volume license installer and this seems to work correctly.
A generated key does allow the XP Installation to continue - and as we know activation is not necessary for the volume license versions of XP.


@RetroSoftwareRepository 
> > I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version
> 
> Ooh nice... also, I'm glad I could be of help for those who have retail XP discs laying around. I happen to have a XP SP2 system builder CD that I know (without the aid of this tool) would have been unusable today because activation actually doesn't work without updating certificates in the OS itself.

Yes! I agree! - Personally, I desire creating a comprehensive tool for both creating keys and activating XP (and other versions of windows/office that use these types of keys)

**Comment by TCB13** [comment]
> (and other versions of windows/office that use these types of keys)

Oh! Jackpot :) I was about to ask about Office 2003!

It would be nice to have a keygen that could generate keys and activation for non-VL versions Windows XP and Office 2003. I also have multiple CDs of retail / OEM versions in different languages of Office that are essentially useless because there's no possible way to active them.

Thank you for your efforts.

**Comment by nononymousse** [comment]
I have a copy of Office 2003 small business and it would be great, i could install it again on a XP machine ^^

btw is there others Microsoft products that use the same Product key scheme as XP ? (like there was during the 95 period)

**Comment by thisgithubaccountaa** [comment]
you are on pc gamer:

https://www.pcgamer.com/a-magical-windows-xp-activation-tool-has-been-hiding-in-plain-sight-on-reddit-for-the-last-year/

**Comment by thisgithubaccountaa** [comment]
> I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version

they specifically mentioned you @Neo-Desktop .

https://www.pcgamer.com/a-magical-windows-xp-activation-tool-has-been-hiding-in-plain-sight-on-reddit-for-the-last-year/#:~:text=We%20may%20never,on%20modern%20Windows.

**Comment by thisgithubaccountaa** [comment]
less than 9 hours after you made that comment, they wrote an article about this.

**Comment by CONIGUERO** [comment]
@Neo-Desktop The YouTuber Enderman is already working on a functional XP Keygen. It can already generate the same VL keys and we're currently in the process of cracking the rest of the Elliptic Curve private keys to support all editions and builds.

https://github.com/Endermanch/XPKeygen

https://github.com/Endermanch/XPKeygen/issues/3#issuecomment-1560476799

**Comment by thepwrtank18** [comment]
So far, the xp_activate32.exe program supports Windows XP Home Edition and Windows Server 2008 R2 x64 Enterprise, from my testing. It doesn't support Office 2003 or Plus! DME, so it's just for Windows.

**Comment by Neo-Desktop** [comment]
@thisgithubaccountaa 
> > I'm working on disassembling `xp_activate32.exe` with the intention of creating an open source version
> 
> they specifically mentioned you @Neo-Desktop .
> 
> https://www.pcgamer.com/a-magical-windows-xp-activation-tool-has-been-hiding-in-plain-sight-on-reddit-for-the-last-year/#:~:text=We%20may%20never,on%20modern%20Windows.

Oh wow, right on! Ya love to see it



@CONIGUERO 
> @Neo-Desktop The YouTuber Enderman is already working on a functional XP Keygen. It can already generate the same VL keys and we're currently in the process of cracking the rest of the Elliptic Curve private keys to support all editions and builds.
> 
> https://github.com/Endermanch/XPKeygen
> 
> [Endermanch/XPKeygen#3 (comment)](https://github.com/Endermanch/XPKeygen/issues/3#issuecomment-1560476799)

Thanks for linking this - it seems that there's a lot more math involved here to make these keygens work. This may be a bit of a stumbling block for me but I'm sure there are people that have a very good understanding of the math involved that might be able to assist



@thepwrtank18 
> So far, the xp_activate32.exe program supports Windows XP Home Edition and Windows Server 2003 R2 x64 Enterprise, from my testing. It doesn't support Office 2003 or Plus! DME, so it's just for Windows.

Ah okay thank you for testing - so it sounds like the activator specifically works with the NT5 product family phone activation. At this point I wouldn't know what the differences are, but thank you for testing this and reporting your results

**Comment by CONIGUERO** [comment]
> Ah okay thank you for testing - so it sounds like the activator specifically works with the NT5 product family phone activation. At this point I wouldn't know what the differences are, but thank you for testing this and reporting your results

@Neo-Desktop It might also just be that the creator didn't bother to check and hardcoded anything else apart from known values to throw that error. After all, the check digit algorithm is the same.

**Comment by dejl** [comment]
> Yes so I just test this on an XP Pro Volume license installer and this seems to work correctly.

I can't seem to get any key generated from this or Enderman's tool to be accepted by the installer,
I've tried the MSDN retail ISO, a VL one I found & editing setupp.ini directly.. am I missing something?

**Comment by dejl** [comment]
Okay I used the proper VL ISO and it works now
for anyone wondering this is the SHA1 of the correct version
66ac289ae27724c5ae17139227cbe78c01eefe40

**Comment by thepwrtank18** [comment]
> @thepwrtank18
> 
> > So far, the xp_activate32.exe program supports Windows XP Home Edition and Windows Server 2003 R2 x64 Enterprise, from my testing. It doesn't support Office 2003 or Plus! DME, so it's just for Windows.
> 
> Ah okay thank you for testing - so it sounds like the activator specifically works with the NT5 product family phone activation. At this point I wouldn't know what the differences are, but thank you for testing this and reporting your results

@Neo-Desktop As an addition, this is what happens when you use an unsupported product:
![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/9dcb61e0-dd7f-494f-9c0e-ea42f96dbea0)

**Comment by thepwrtank18** [comment]
Are there chances this could be an internal keygen from Microsoft themselves? It's probably low, but most keygens have stuff like "MADE BY X-FORCE" and have ear deafening music.

**Comment by Neo-Desktop** [comment]
@thepwrtank18 
> Are there chances this could be an internal keygen from Microsoft themselves? It's probably low, but most keygens have stuff like "MADE BY X-FORCE" and have ear deafening music.

No, this tool has a whole separate dialog in Cyrillic and it breaks some best practices in the code that very likely wouldn't fly at Microsoft.

I'm fairly positive any Microsoft specific tool for generating phone activation confirmation ids would be some kind of client-server (like asp.net) webapp that integrates with the automated phone (IVR-based) activation system

**Comment by vrubleg** [comment]
The xp_activate32.exe's timestamp is 0x5A8711DB, it's 2018/02/16 19:16:11. So, it's 5 years old already.

**Comment by thepwrtank18** [comment]
https://forums.mydigitallife.net/threads/is-there-any-way-to-crack-decrypt-the-winxp-consumer-activation-system-to-generate-activation-ids.80133/page-4#post-1791193
This guy says he has the source.

**Comment by Hukuma1** [comment]
He just posted it. Grab it while you can.

**Comment by Neo-Desktop** [comment]
@Hukuma1  @thepwrtank18 
> He just posted it. Grab it while you can.
> 
> https://forums.mydigitallife.net/threads/is-there-any-way-to-crack-decrypt-the-winxp-consumer-activation-system-to-generate-activation-ids.80133/page-4#post-1791266

Got it. Thanks

**Comment by thepwrtank18** [comment]
for the love of Bill "Microsoft" Gates, upload faster
![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/769fdd82-58dc-4c54-ba8b-a26ed9f17f8f)

**Comment by thepwrtank18** [comment]
https://archive.org/details/xp_activate32_src

**Comment by thepwrtank18** [comment]
Time to re-download Visual Studio, since I have to reinstall that after I reset my PC.

**Comment by CONIGUERO** [comment]
I made a git repo for it here: https://github.com/CONIGUERO/xpcid/blob/main/xp_activate32.c

**Comment by thepwrtank18** [comment]
When I get VS reinstalled, I'll update it to be something you can drop into VS to build.

**Comment by thepwrtank18** [comment]
![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/177514ad-5eb1-45d4-9347-d41794fa3bce)
Has anyone fixed this?

**Comment by TCB13** [comment]
> ![image](https://user-images.githubusercontent.com/57580668/241426437-177514ad-5eb1-45d4-9347-d41794fa3bce.png) Has anyone fixed this?

Click yes, proceed with phone activation using xp_activate32.

**Comment by thepwrtank18** [comment]
> > ![image](https://user-images.githubusercontent.com/57580668/241426437-177514ad-5eb1-45d4-9347-d41794fa3bce.png) Has anyone fixed this?
> 
> Click yes, proceed with phone activation using xp_activate32.


It just says it's already activated.

**Comment by vrubleg** [comment]
Boot into "Safe Mode". Press F8 during booting. Run:
```
rundll32.exe syssetup,SetupOobeBnk
```

**Comment by thepwrtank18** [comment]
Tried, doesn't do anything.

**Comment by CONIGUERO** [comment]
> Tried, doesn't do anything.

In Safe Mode, delete C:\Windows\System32\wpa.dbl and wpa.bak

**Comment by thepwrtank18** [comment]
> > Tried, doesn't do anything.
> 
> In Safe Mode, delete C:\Windows\System32\wpa.dbl and wpa.bak

The "already activated" issue is gone, but...

![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/ae81e53f-e355-4108-bdb5-3a462278088a)

there's no installation ID.

**Comment by TCB13** [comment]
> there's no installation ID.

Maybe you can change the product key to another one and check if the ID appears. It seems like something messed up with your system. Maybe a previous attempt at cracking it?

**Comment by thepwrtank18** [comment]
> > there's no installation ID.
> 
> Maybe you can change the product key to another one and check if the ID appears. It seems like something messed up with your system. Maybe a previous attempt at cracking it?

Changed product key, nothing. All I used to "crack" was xp_activate32.exe.

**Comment by Neo-Desktop** [comment]
@thepwrtank18 
> > > there's no installation ID.
> > 
> > 
> > Maybe you can change the product key to another one and check if the ID appears. It seems like something messed up with your system. Maybe a previous attempt at cracking it?
> 
> Changed product key, nothing. All I used to "crack" was xp_activate32.exe.
> 
> Edit: It looks like changing the product key fixes the issue (gets you back in explorer), but only after a restart. From there, you have to go to msoobe /a and redo the process. After that, it's actually for real this time activated.

Sounds like you got a "unlucky" key, I'm not entirely sure what causes it but it has been documented in various threads and on the [Endermanch/XPKeygen](https://github.com/Endermanch/XPKeygen) repository

**Comment by thepwrtank18** [comment]
For documentation, here it is: PW6PT-TCGBR-HKTCT-GKKY6-QGK86

**Comment by thisgithubaccountaa** [comment]
the next iteration of ChatGPT will be able to crack Windows XP because of this.  ( :

**Comment by thepwrtank18** [comment]
> the next iteration of ChatGPT will be able to crack Windows XP because of this. ( :

Eh, @endermanch's video showed that you have to push ChatGPT a lot to get Windows 95 to make valid keys.

**Comment by thepwrtank18** [comment]
I'm making a list of every Windows XP edition and the keys associated with them that work. Here's what it looks like now:
![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/d8101823-ab66-469e-a32c-cf33aacef0a7)

**Comment by thepwrtank18** [comment]
Anyone have a Professional x64 Edition OEM key? I can't find any, but I know it exists.

**Comment by thepwrtank18** [comment]
Here it is (Windows XP only, other NT 5 will be added)

| Operating System | Edition                | License | Key                                                                             | Notes                                                                                                   |
| ---------------- | ---------------------- | ------- | ------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- |
| Windows XP       | Starter Edition        | OEM     | JWDWF-8V6YT-R98XR-3YMVQ-HBMPW                                                   |                                                                                                         |
| Windows XP       | Home Edition (+ N)     | Retail  | XMCM6-DKYCQ-2BHQH-4PCHR-TBJCR                                                   |                                                                                                         |
| Windows XP       | Home Edition (+ N)     | OEM     | G7BR9-8QV29-3QFHP-F84WG-X9PYQ                                                   | N edition created via SETUPP.ini hack. If you have a real version, please upload it for proper testing! |
| Windows XP       | Home Edition K         | Retail  | W8F6Q-HM3JB-2XRHD-7Q92J-XKY6W                                                   |                                                                                                         |
| Windows XP       | Home Edition K         | OEM     | [Doesn't seem to exist. If you have this, please give the key and/or ISO!]      |                                                                                                         |
| Windows XP       | Home Edition KN        | Retail  | M9D9J-2TQV2-FBJQP-2M8G8-DGQ26                                                   |                                                                                                         |
| Windows XP       | Home Edition KN        | OEM     | [Doesn't seem to exist. If you have this, please give the key and/or ISO!]      |                                                                                                         |
| Windows XP       | Professional (+ N)     | Retail  | CD87T-HFP4C-V7X7H-8VY68-W7D7M                                                   |                                                                                                         |
| Windows XP       | Professional (+ N)     | OEM     | XJM6Q-BQ8HW-T6DFB-Y934T-YD4YT                                                   | N edition created via SETUPP.ini hack. If you have a real version, please upload it for proper testing! |
| Windows XP       | Professional (+ N)     | Volume  | M6TF9-8XQ2M-YQK9F-7TBB2-XGG88                                                   |                                                                                                         |
| Windows XP       | Professional K         | Retail  | GXYPJ-4W67Q-6273F-QJ7FR-9MW4M                                                   |                                                                                                         |
| Windows XP       | Professional K         | OEM     | B4B4Y-4XCXM-C7P82-HMJ6D-9HBFG                                                   | Created via SETUPP.ini hack. If you have a real version, please upload it for proper testing!           |
| Windows XP       | Professional K         | Volume  | [Doesn't seem to exist. If you have this, please give the key and/or ISO!]      |                                                                                                         |
| Windows XP       | Professional KN        | Retail  | BTJQ7-3VGFD-GMHRJ-MTGQX-RTVXQ                                                   |                                                                                                         |
| Windows XP       | Professional KN        | OEM     | [Doesn't seem to exist. If you have this, please give the key and/or ISO!]      |                                                                                                         |
| Windows XP       | Professional KN        | Volume  | [Doesn't seem to exist. If you have this, please give the key and/or ISO!]      |                                                                                                         |
| Windows XP       | Professional x64       | Retail  | B2RBK-7KPT9-4JP6X-QQFWM-PJD6G                                                   |                                                                                                         |
| Windows XP       | Professional x64       | OEM     | [Exists, but can't be found. If you have this, please give the key and/or ISO!] |                                                                                                         |
| Windows XP       | Professional x64       | Volume  | B66VY-4D94T-TPPD4-43F72-8X4FY                                                   |                                                                                                         |
| Windows XP       | Tablet PC (2002-05)    | OEM     | KY6GY-YHDPV-RMY9G-PHG6P-2F9Y3                                                   |                                                                                                         |
| Windows XP       | Tablet PC (2002-05)    | Volume  | KT3Q2-F3JJK-WGHVH-4TTP2-X3GRD                                                   |                                                                                                         |
| Windows XP       | Media Center (2002-05) | OEM     | C4BH3-P4J7W-9MT6X-PGKC8-J4JTM                                                   |                                                                                                         |

**Comment by daf-scripts** [comment]
Hey everyone wanted to jump in here with a few of my findings:

Steps I took:

1. Downloaded all XP, Server 2003, and Home Server isos from https://files.dog/MSDN/
2. Utilize pycdlib (https://github.com/clalancette/pycdlib ) to open each iso file and extract and files containing "pid"
3. Expand the compressed files ending in "_"
4. This effort can be found here https://1drv.ms/u/s!AlPIFTdGZCGGbSq5Ax3IXJpu33Q?e=b3ppx7
5. Next utilize EllipticCurveTool (from: https://forums.mydigitallife.net/threads/elliptic-curve-product-keys.84765/page-2#post-1791162 ) to convert all the de-duplicated dll files found above into the ks2 format
6. These ks2 files can be found https://1drv.ms/u/s!AlPIFTdGZCGGbMOLMWysalJCmZw?e=TTGiQz
7. Next take all the dll files and use this (https://www.nirsoft.net/utils/resources_extract.html) to extract the BINK resources
8. Utilize the BINK decoder script here (https://github.com/Endermanch/XPKeygen/blob/main/BINKReader.py) to output BINK data in txt format
9. You can find the BINK files and the decoded data in txt format here https://1drv.ms/u/s!AlPIFTdGZCGGa1vSUlp3MnVQeK0?e=cr0qV2

I also took a stab at modifying the sage script to input the ks2 files and output the job file for Mr. HAANDI's ECDLP Solver v0.2a (can be downloaded from https://dl.malwarewatch.org/software/advanced/ecc-research-tools/ )
You can find this converter script here: https://1drv.ms/u/s!AlPIFTdGZCGGahta_OnkGk35f-E?e=U4XiVu 
This has only been tested that it outputs numbers not that the numbers are correct :P

If you download all of the above you will have 24 BINKs to play with and find k and n values. Then it is just a matter of matching up the BINK with a specific iso.

Let me know if there are other file names I should look at or other iso directories!

**Comment by tothi** [comment]
> The xp_activate32.exe's timestamp is 0x5A8711DB, it's 2018/02/16 19:16:11. So, it's 5 years old already.

Not sure (it can be faked), but it has been seen on [virustotal 3 yrs ago](https://twitter.com/an0n_r0/status/1662432364142755841). So it is min 3 yrs old.

**Comment by vrubleg** [comment]
It makes no sense to fake it, and there is already the source code with file dates that match with build time.

**Comment by tothi** [comment]
> It makes no sense to fake it, and there is already the source code with file dates that match with build time.

totally agree, just wanted to provide a 100% reliable proof about it is not new. (that's why I said "min").

**Comment by andika207** [comment]
my VL ISO has never required activation. LOL.

<a href='https://postimages.org/' target='_blank'><img src='https://i.postimg.cc/c4Tsgpph/screenshot-1.png' border='0' alt='screenshot-1'/></a>

**Comment by Endermanch** [comment]
I did attempt to reverse the Confirmation ID generator. I did not get very far, I dropped it. In case anyone wants, I can send you my IDA database so you could continue reversing it.

![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/44542704/7993f282-0923-4669-abd0-a79cd011cb52)

**Comment by thepwrtank18** [comment]
> my VL ISO has never required activation. LOL.
> 
> [![screenshot-1](https://camo.githubusercontent.com/9c44cee29bc42ac1ae20c1c54fdaf48985c59cb019591b6202ce14c04c921c52/68747470733a2f2f692e706f7374696d672e63632f63345473677070682f73637265656e73686f742d312e706e67)](https://postimages.org/)

I'm still perplexed that Microsoft just made an open "hey, free Windows XP with no activation required!" and didn't put SOME protection on the Volume License versions.

**Comment by scerazy** [comment]
VL never did!

**Comment by thepwrtank18** [comment]
> These were the times of really slow internet & different world

Good point, forgot that these were the days of Limewire where you had to take a gamble if the download you wanted was actually what you wanted.

**Comment by thepwrtank18** [comment]
Getting started with Windows Server 2003.
![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/26e758a8-6367-48a9-9ace-cfd0a8ef8438)

**Comment by Endermanch** [comment]
> my VL ISO has never required activation. LOL.
> 
> [![screenshot-1](https://camo.githubusercontent.com/9c44cee29bc42ac1ae20c1c54fdaf48985c59cb019591b6202ce14c04c921c52/68747470733a2f2f692e706f7374696d672e63632f63345473677070682f73637265656e73686f742d312e706e67)](https://postimages.org/)

This is the idea. Volume licensing keys are created so that multiple or even many computers can use the same key without any problems. OEM keys, on the other hand, have one-to-one correspondence with the computers they've been installed onto, so they need to be "proven genuine". OEM activation has more gimmicks to it than you think, and phone activation isn't the only one.

**Comment by smartphonecomputerundmehr** [comment]
I tested Office XP and Windows XP Media Center Edition 2005
Office didn´t work, MCE did

**Comment by thepwrtank18** [comment]
I should note that everything I'm putting in the table was either tested to work with xp_activate32.exe, or didn't need activation.

**Comment by thepwrtank18** [comment]
Turns out there's a lot more derivatives of XP then I thought.
Here's the list:
* XP
* POSReady 2009
* Fundamentals for Legacy PC's
* Server 2003
* Server 2003 R2
* Compute Cluster Server
* Small Business Server
* Home Server
* Server for Embedded Systems
* Server for Embedded Systems R2
* Storage Server

This doesn't include their individual editions.

**Comment by Endermanch** [comment]
> Turns out there's a lot more derivatives of XP then I thought.
> Here's the list:
> * XP
> * POSReady 2009
> * Fundamentals for Legacy PC's
> * Server 2003
> * Server 2003 R2
> * Compute Cluster Server
> * Small Business Server
> * Home Server
> * Server for Embedded Systems
> * Server for Embedded Systems R2
> * Storage Server
> 
> This doesn't include their individual editions.

POSReady must be the one I'm looking forward to the most. Please check that

**Comment by thepwrtank18** [comment]
Installing POSReady now, just gotta wait for it to finish so I can test.

**Comment by thepwrtank18** [comment]
![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/f6ea72d7-7c86-47ab-8312-d47709ea6b48)

**Comment by thepwrtank18** [comment]
![image](https://github.com/Neo-Desktop/WindowsXPKg/assets/57580668/dca3ad52-e62a-47fb-871e-750bc01a0037)
There we go.

**Comment by thepwrtank18** [comment]
POSReady 2009 looks like a "no activation needed" product.

**Comment by thepwrtank18** [comment]
The almost-final table:

| Operating System                     | Edition                  | License | Key                                                                               | Notes                                                                                                   |
| ------------------------------------ | ------------------------ | ------- | --------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- |
| Windows XP                           | Starter Edition          | OEM     | `JWDWF-8V6YT-R98XR-3YMVQ-HBMPW`                                                   |                                                                                                         |
| Windows XP                           | Home Edition (+ N)       | Retail  | `XMCM6-DKYCQ-2BHQH-4PCHR-TBJCR`                                                   |                                                                                                         |
| Windows XP                           | Home Edition (+ N/ULCPC) | OEM     | `G7BR9-8QV29-3QFHP-F84WG-X9PYQ`                                                   | N edition created via SETUPP.ini hack. If you have a real version, please upload it for proper testing! |
| Windows XP                           | Home Edition K           | Retail  | `W8F6Q-HM3JB-2XRHD-7Q92J-XKY6W`                                                   |                                                                                                         |
| Windows XP                           | Home Edition K           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows XP                           | Home Edition KN          | Retail  | `M9D9J-2TQV2-FBJQP-2M8G8-DGQ26`                                                   |                                                                                                         |
| Windows XP                           | Home Edition KN          | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows XP                           | Professional (+ N)       | Retail  | `CD87T-HFP4C-V7X7H-8VY68-W7D7M`                                                   |                                                                                                         |
| Windows XP                           | Professional (+ N)       | OEM     | `XJM6Q-BQ8HW-T6DFB-Y934T-YD4YT`                                                   | N edition created via SETUPP.ini hack. If you have a real version, please upload it for proper testing! |
| Windows XP                           | Professional (+ N)       | Volume  | `M6TF9-8XQ2M-YQK9F-7TBB2-XGG88`                                                   |                                                                                                         |
| Windows XP                           | Professional K           | Retail  | `GXYPJ-4W67Q-6273F-QJ7FR-9MW4M`                                                   |                                                                                                         |
| Windows XP                           | Professional K           | OEM     | `B4B4Y-4XCXM-C7P82-HMJ6D-9HBFG`                                                   | Created via SETUPP.ini hack. If you have a real version, please upload it for proper testing!           |
| Windows XP                           | Professional K           | Volume  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows XP                           | Professional KN          | Retail  | `BTJQ7-3VGFD-GMHRJ-MTGQX-RTVXQ`                                                   |                                                                                                         |
| Windows XP                           | Professional KN          | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows XP                           | Professional KN          | Volume  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows XP                           | Professional x64         | Retail  | `B2RBK-7KPT9-4JP6X-QQFWM-PJD6G`                                                   |                                                                                                         |
| Windows XP                           | Professional x64         | OEM     | Exists, but can't be found. If you have this, please give the key and/or ISO!     |                                                                                                         |
| Windows XP                           | Professional x64         | Volume  | `B66VY-4D94T-TPPD4-43F72-8X4FY`                                                   |                                                                                                         |
| Windows XP                           | Tablet PC (2002-05)      | OEM     | `KY6GY-YHDPV-RMY9G-PHG6P-2F9Y3`                                                   |                                                                                                         |
| Windows XP                           | Tablet PC (2002-05)      | Volume  | `KT3Q2-F3JJK-WGHVH-4TTP2-X3GRD`                                                   |                                                                                                         |
| Windows XP                           | Media Center (2002-05)   | OEM     | `C4BH3-P4J7W-9MT6X-PGKC8-J4JTM`                                                   |                                                                                                         |
| Windows Embedded POSReady 2009       | [N/A]                    | OEM     | `D4GQ7-HG48K-7YP6R-6JM4C-33FMQ`                                                   |                                                                                                         |
| Windows Fundamentals for Legacy PC's | [N/A]                    | Volume  | `JWTG2-3CB49-QKWCW-R9DW9-MCJMQ`                                                   |                                                                                                         |
| Windows Server 2003                  | Web                      | Retail  | `D42X8-7MWXD-M4B76-MKYP7-CW9FD`                                                   |                                                                                                         |
| Windows Server 2003                  | Web                      | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003                  | Web                      | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   |                                                                                                         |
| Windows Server 2003                  | Standard                 | Retail  | `M6RJ9-TBJH3-9DDXM-4VX9Q-K8M8M`                                                   |                                                                                                         |
| Windows Server 2003                  | Standard                 | OEM     | `KM63V-GW6H6-WY4BP-HMCK4-G7CKJ`                                                   |                                                                                                         |
| Windows Server 2003                  | Standard                 | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   | Duplicate of Web, still works.                                                                          |
| Windows Server 2003                  | Standard x64             | Retail  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
| Windows Server 2003                  | Standard x64             | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003                  | Standard x64             | Volume  | `BXD88-QGH3V-XG44M-MDXGW-CBYQY`                                                   |                                                                                                         |
| Windows Server 2003                  | Enterprise               | Retail  | `QW32K-48T2T-3D2PJ-DXBWY-C6WRJ`                                                   |                                                                                                         |
| Windows Server 2003                  | Enterprise               | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003                  | Enterprise               | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   | Duplicate of Web, still works.                                                                          |
| Windows Server 2003                  | Enterprise x64           | Retail  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
| Windows Server 2003                  | Enterprise x64           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003                  | Enterprise x64           | Volume  | `BXD88-QGH3V-XG44M-MDXGW-CBYQY`                                                   | Duplicate of Standard x64, still works.                                                                 |
| Windows Server 2003                  | Datacenter               | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003                  | Datacenter               | OEM     | `QW32K-48T2T-3D2PJ-DXBWY-C6WRJ`                                                   | Duplicate of Enteprise Retail, still works.                                                             |
| Windows Server 2003                  | Datacenter               | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   | Duplicate of Web, still works.                                                                          |
| Windows Server 2003                  | Compute Cluster          | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003                  | Compute Cluster          | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003                  | Compute Cluster          | Volume  | `BXD88-QGH3V-XG44M-MDXGW-CBYQY`                                                   | Duplicate of Standard x64, still works.                                                                 |
| Windows Server 2003 R2               | Web                      | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Web                      | OEM     | `GM8KD-GB7JY-QGQYP-XRV74-RT728`                                                   | Untestable as I don't have this version. If you do, please give it!                                     |
| Windows Server 2003 R2               | Web                      | Volume  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Standard                 | Retail  | `QKDCQ-TP2JM-G4MDG-VR6F2-P9C48`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Standard                 | OEM     | `PWBJC-22697-D4CVH-FCJWW-DTF9J`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Standard                 | Volume  | `HDXR2-62748-6XG6Y-4BT6T-R6H43`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Standard KN              | Retail  | `RHVXX-TB66G-4PMT2-JVWKM-X7XRQ`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Standard KN              | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Standard KN              | Volume  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
| Windows Server 2003 R2               | Standard x64             | Retail  | `P4WJG-WK3W7-3HM8W-RWHCK-8JTRY`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Standard x64             | OEM     | `XCP6P-7WVXP-F8FQ4-JV6CD-6XV28`                                                   | Untestable as I don't have this version. If you do, please give it!                                     |
| Windows Server 2003 R2               | Standard x64             | Volume  | `B76XF-FCR39-XBGFF-YMBFH-G6PMT`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Enterprise               | Retail  | `BMCM6-WYKBH-2CTVY-KKKFM-RTP6G`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Enterprise               | OEM     | `XHPV3-PTCWJ-7Y94F-Q6BVH-J849J`                                                   | Untestable as I don't have this version. If you do, please give it!                                     |
| Windows Server 2003 R2               | Enterprise               | Volume  | `HDXR2-62748-6XG6Y-4BT6T-R6H43`                                                   | Duplicate of Standard, still works.                                                                     |
| Windows Server 2003 R2               | Enterprise KN            | Retail  | `TBK82-HCT7F-XKXRC-YWK2F-GHG4Q`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Enterprise KN            | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Enterprise KN            | Volume  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
| Windows Server 2003 R2               | Enterprise x64           | Retail  | `VYTMJ-8K9XR-FK63H-6VBXX-8TJJ6`                                                   |                                                                                                         |
| Windows Server 2003 R2               | Enterprise x64           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Enterprise x64           | Volume  | `B76XF-FCR39-XBGFF-YMBFH-G6PMT`                                                   | Duplicate of Standard x64, still works.                                                                 |
| Windows Server 2003 R2               | Datacenter               | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Datacenter               | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Datacenter               | Volume  | `HDXR2-62748-6XG6Y-4BT6T-R6H43`                                                   | Duplicate of Standard, still works.                                                                     |
| Windows Server 2003 R2               | Datacenter x64           | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Datacenter x64           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
| Windows Server 2003 R2               | Datacenter x64           | Volume  | `B76XF-FCR39-XBGFF-YMBFH-G6PMT`                                                   | Duplicate of Standard x64, still works.                                                                 |

Just gotta deal with the weird one-off's like Home Server and stuff.

**Comment by thepwrtank18** [comment]
https://gist.github.com/thepwrtank18/4456b1a4676a26c6ef25b8e8b70e26d7

**Comment by ClackKent** [comment]
Windows xp x64 retail not support

**Comment by ClackKent** [comment]
Windows xp x64 retail not support

**Comment by thepwrtank18** [comment]
> Windows xp x64 retail not support

Worked on my testing. Key?

**Comment by andika207** [comment]
> Turns out there's a lot more derivatives of XP then I thought.

@thepwrtank18  -Windows Embedded Standard 2009 Evaluation

**Comment by WitherOrNot** [comment]
Embedded POSReady 2009 Key: `KFF8R-TMHF4-GKT2P-KGDHH-FXPMB`

**Comment by ward201185** [comment]
> @thepwrtank18
> 
> > > > there's no installation ID.
> > > 
> > > 
> > > Maybe you can change the product key to another one and check if the ID appears. It seems like something messed up with your system. Maybe a previous attempt at cracking it?
> > 
> > 
> > Changed product key, nothing. All I used to "crack" was xp_activate32.exe.
> > Edit: It looks like changing the product key fixes the issue (gets you back in explorer), but only after a restart. From there, you have to go to msoobe /a and redo the process. After that, it's actually for real this time activated.
> 
> Sounds like you got a "unlucky" key, I'm not entirely sure what causes it but it has been documented in various threads and on the [Endermanch/XPKeygen](https://github.com/Endermanch/XPKeygen) repository

i may have another issue similar to this.
this key was generated via xpkey.exe from 2 years ago.
it is for home edition retail and causes odd behaviour.
the key is accepted during setup but during the oobe phase, it goes straight to the login screen and skips everything.
im then greeted with the same activation error that thepwrtank18 got.
but after clicking yes, the gui barely loads and the activation box wont load.
key: PP3JK-T7FB8-BGPMB-MWKQW-XXD8J
Product ID: 76477-004-0674031-21418
also, if you had already entered a product key before and you changed the product key into the one above, it will accept it until you do a restart and its brought back to the previous product key before you changed it.
it might've also overwritten wpa.dbl with wpa.bak as i dont see wpa.bak after restarting.
this issue is replicatable.
i just wanted to bring this up if it strikes a clue or makes identifying the issue alot easier.

**Comment by thepwrtank18** [comment]
> but after clicking yes, the gui barely loads and the activation box wont load.

Same here, perfect replication of the bug.

**Comment by WitherOrNot** [comment]
@ward201185 

This key probably exhibits strange behavior because its site code (004) is not in the allowed ranges specified by DPCDLL.DLL.

These ranges seem to have something to do with generating correct installation IDs from keys. I opened #15 to look into this further.

The allowed PID ranges for your ISO can be seen in this JSON list:

```json
[
    {
        "index": 0,
        "bink": "0000002B",
        "pid_range": [
            119,
            119
        ],
        "type": "OEM",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 1,
        "bink": "0000002B",
        "pid_range": [
            120,
            169
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 2,
        "bink": "0000002B",
        "pid_range": [
            400,
            665
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 3,
        "bink": "0000002B",
        "pid_range": [
            667,
            699
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 4,
        "bink": "0000002B",
        "pid_range": [
            700,
            754
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 16,
        "bink": "0000002A",
        "pid_range": [
            5,
            85
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 17,
        "bink": "0000002A",
        "pid_range": [
            337,
            359
        ],
        "type": "Tablet",
        "days_to_activate": 60,
        "days_evaluation": 2147483647
    },
    {
        "index": 24,
        "bink": "0000002A",
        "pid_range": [
            755,
            779
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 43,
        "bink": "0000002B",
        "pid_range": [
            85,
            85
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 44,
        "bink": "0000002B",
        "pid_range": [
            86,
            95
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 45,
        "bink": "0000002B",
        "pid_range": [
            96,
            105
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 46,
        "bink": "0000002A",
        "pid_range": [
            106,
            106
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 47,
        "bink": "0000002A",
        "pid_range": [
            187,
            187
        ],
        "type": "Tablet",
        "days_to_activate": 60,
        "days_evaluation": 2147483647
    },
    {
        "index": 48,
        "bink": "0000002A",
        "pid_range": [
            188,
            188
        ],
        "type": "Tablet",
        "days_to_activate": 60,
        "days_evaluation": 2147483647
    },
    {
        "index": 49,
        "bink": "0000002B",
        "pid_range": [
            170,
            186
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 50,
        "bink": "0000002B",
        "pid_range": [
            189,
            189
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 51,
        "bink": "0000002B",
        "pid_range": [
            190,
            190
        ],
        "type": "OEM",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 52,
        "bink": "0000002B",
        "pid_range": [
            191,
            191
        ],
        "type": "OEM",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 62,
        "bink": "0000002B",
        "pid_range": [
            107,
            118
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 63,
        "bink": "0000002B",
        "pid_range": [
            192,
            336
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 64,
        "bink": "0000002B",
        "pid_range": [
            755,
            899
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 65,
        "bink": "0000002B",
        "pid_range": [
            921,
            979
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 72,
        "bink": "0000002A",
        "pid_range": [
            362,
            376
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 73,
        "bink": "0000002A",
        "pid_range": [
            377,
            377
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    },
    {
        "index": 74,
        "bink": "0000002A",
        "pid_range": [
            984,
            984
        ],
        "type": "Retail",
        "days_to_activate": 30,
        "days_evaluation": 2147483647
    }
]
```

**Comment by RetroSoftwareRepository** [comment]
> The almost-final table:
> 
> 
> 
> | Operating System                     | Edition                  | License | Key                                                                               | Notes                                                                                                   |
> 
> | ------------------------------------ | ------------------------ | ------- | --------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- |
> 
> | Windows XP                           | Starter Edition          | OEM     | `JWDWF-8V6YT-R98XR-3YMVQ-HBMPW`                                                   |                                                                                                         |
> 
> | Windows XP                           | Home Edition (+ N)       | Retail  | `XMCM6-DKYCQ-2BHQH-4PCHR-TBJCR`                                                   |                                                                                                         |
> 
> | Windows XP                           | Home Edition (+ N/ULCPC) | OEM     | `G7BR9-8QV29-3QFHP-F84WG-X9PYQ`                                                   | N edition created via SETUPP.ini hack. If you have a real version, please upload it for proper testing! |
> 
> | Windows XP                           | Home Edition K           | Retail  | `W8F6Q-HM3JB-2XRHD-7Q92J-XKY6W`                                                   |                                                                                                         |
> 
> | Windows XP                           | Home Edition K           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows XP                           | Home Edition KN          | Retail  | `M9D9J-2TQV2-FBJQP-2M8G8-DGQ26`                                                   |                                                                                                         |
> 
> | Windows XP                           | Home Edition KN          | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows XP                           | Professional (+ N)       | Retail  | `CD87T-HFP4C-V7X7H-8VY68-W7D7M`                                                   |                                                                                                         |
> 
> | Windows XP                           | Professional (+ N)       | OEM     | `XJM6Q-BQ8HW-T6DFB-Y934T-YD4YT`                                                   | N edition created via SETUPP.ini hack. If you have a real version, please upload it for proper testing! |
> 
> | Windows XP                           | Professional (+ N)       | Volume  | `M6TF9-8XQ2M-YQK9F-7TBB2-XGG88`                                                   |                                                                                                         |
> 
> | Windows XP                           | Professional K           | Retail  | `GXYPJ-4W67Q-6273F-QJ7FR-9MW4M`                                                   |                                                                                                         |
> 
> | Windows XP                           | Professional K           | OEM     | `B4B4Y-4XCXM-C7P82-HMJ6D-9HBFG`                                                   | Created via SETUPP.ini hack. If you have a real version, please upload it for proper testing!           |
> 
> | Windows XP                           | Professional K           | Volume  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows XP                           | Professional KN          | Retail  | `BTJQ7-3VGFD-GMHRJ-MTGQX-RTVXQ`                                                   |                                                                                                         |
> 
> | Windows XP                           | Professional KN          | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows XP                           | Professional KN          | Volume  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows XP                           | Professional x64         | Retail  | `B2RBK-7KPT9-4JP6X-QQFWM-PJD6G`                                                   |                                                                                                         |
> 
> | Windows XP                           | Professional x64         | OEM     | Exists, but can't be found. If you have this, please give the key and/or ISO!     |                                                                                                         |
> 
> | Windows XP                           | Professional x64         | Volume  | `B66VY-4D94T-TPPD4-43F72-8X4FY`                                                   |                                                                                                         |
> 
> | Windows XP                           | Tablet PC (2002-05)      | OEM     | `KY6GY-YHDPV-RMY9G-PHG6P-2F9Y3`                                                   |                                                                                                         |
> 
> | Windows XP                           | Tablet PC (2002-05)      | Volume  | `KT3Q2-F3JJK-WGHVH-4TTP2-X3GRD`                                                   |                                                                                                         |
> 
> | Windows XP                           | Media Center (2002-05)   | OEM     | `C4BH3-P4J7W-9MT6X-PGKC8-J4JTM`                                                   |                                                                                                         |
> 
> | Windows Embedded POSReady 2009       | [N/A]                    | OEM     | `D4GQ7-HG48K-7YP6R-6JM4C-33FMQ`                                                   |                                                                                                         |
> 
> | Windows Fundamentals for Legacy PC's | [N/A]                    | Volume  | `JWTG2-3CB49-QKWCW-R9DW9-MCJMQ`                                                   |                                                                                                         |
> 
> | Windows Server 2003                  | Web                      | Retail  | `D42X8-7MWXD-M4B76-MKYP7-CW9FD`                                                   |                                                                                                         |
> 
> | Windows Server 2003                  | Web                      | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003                  | Web                      | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   |                                                                                                         |
> 
> | Windows Server 2003                  | Standard                 | Retail  | `M6RJ9-TBJH3-9DDXM-4VX9Q-K8M8M`                                                   |                                                                                                         |
> 
> | Windows Server 2003                  | Standard                 | OEM     | `KM63V-GW6H6-WY4BP-HMCK4-G7CKJ`                                                   |                                                                                                         |
> 
> | Windows Server 2003                  | Standard                 | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   | Duplicate of Web, still works.                                                                          |
> 
> | Windows Server 2003                  | Standard x64             | Retail  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
> 
> | Windows Server 2003                  | Standard x64             | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003                  | Standard x64             | Volume  | `BXD88-QGH3V-XG44M-MDXGW-CBYQY`                                                   |                                                                                                         |
> 
> | Windows Server 2003                  | Enterprise               | Retail  | `QW32K-48T2T-3D2PJ-DXBWY-C6WRJ`                                                   |                                                                                                         |
> 
> | Windows Server 2003                  | Enterprise               | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003                  | Enterprise               | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   | Duplicate of Web, still works.                                                                          |
> 
> | Windows Server 2003                  | Enterprise x64           | Retail  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
> 
> | Windows Server 2003                  | Enterprise x64           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003                  | Enterprise x64           | Volume  | `BXD88-QGH3V-XG44M-MDXGW-CBYQY`                                                   | Duplicate of Standard x64, still works.                                                                 |
> 
> | Windows Server 2003                  | Datacenter               | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003                  | Datacenter               | OEM     | `QW32K-48T2T-3D2PJ-DXBWY-C6WRJ`                                                   | Duplicate of Enteprise Retail, still works.                                                             |
> 
> | Windows Server 2003                  | Datacenter               | Volume  | `JB88F-WT2Q3-DPXTT-Y8GHG-7YYQY`                                                   | Duplicate of Web, still works.                                                                          |
> 
> | Windows Server 2003                  | Compute Cluster          | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003                  | Compute Cluster          | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003                  | Compute Cluster          | Volume  | `BXD88-QGH3V-XG44M-MDXGW-CBYQY`                                                   | Duplicate of Standard x64, still works.                                                                 |
> 
> | Windows Server 2003 R2               | Web                      | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Web                      | OEM     | `GM8KD-GB7JY-QGQYP-XRV74-RT728`                                                   | Untestable as I don't have this version. If you do, please give it!                                     |
> 
> | Windows Server 2003 R2               | Web                      | Volume  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard                 | Retail  | `QKDCQ-TP2JM-G4MDG-VR6F2-P9C48`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard                 | OEM     | `PWBJC-22697-D4CVH-FCJWW-DTF9J`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard                 | Volume  | `HDXR2-62748-6XG6Y-4BT6T-R6H43`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard KN              | Retail  | `RHVXX-TB66G-4PMT2-JVWKM-X7XRQ`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard KN              | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard KN              | Volume  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard x64             | Retail  | `P4WJG-WK3W7-3HM8W-RWHCK-8JTRY`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Standard x64             | OEM     | `XCP6P-7WVXP-F8FQ4-JV6CD-6XV28`                                                   | Untestable as I don't have this version. If you do, please give it!                                     |
> 
> | Windows Server 2003 R2               | Standard x64             | Volume  | `B76XF-FCR39-XBGFF-YMBFH-G6PMT`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Enterprise               | Retail  | `BMCM6-WYKBH-2CTVY-KKKFM-RTP6G`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Enterprise               | OEM     | `XHPV3-PTCWJ-7Y94F-Q6BVH-J849J`                                                   | Untestable as I don't have this version. If you do, please give it!                                     |
> 
> | Windows Server 2003 R2               | Enterprise               | Volume  | `HDXR2-62748-6XG6Y-4BT6T-R6H43`                                                   | Duplicate of Standard, still works.                                                                     |
> 
> | Windows Server 2003 R2               | Enterprise KN            | Retail  | `TBK82-HCT7F-XKXRC-YWK2F-GHG4Q`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Enterprise KN            | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Enterprise KN            | Volume  | Can't find key. If you have this, please give the key!                            |                                                                                                         |
> 
> | Windows Server 2003 R2               | Enterprise x64           | Retail  | `VYTMJ-8K9XR-FK63H-6VBXX-8TJJ6`                                                   |                                                                                                         |
> 
> | Windows Server 2003 R2               | Enterprise x64           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Enterprise x64           | Volume  | `B76XF-FCR39-XBGFF-YMBFH-G6PMT`                                                   | Duplicate of Standard x64, still works.                                                                 |
> 
> | Windows Server 2003 R2               | Datacenter               | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Datacenter               | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Datacenter               | Volume  | `HDXR2-62748-6XG6Y-4BT6T-R6H43`                                                   | Duplicate of Standard, still works.                                                                     |
> 
> | Windows Server 2003 R2               | Datacenter x64           | Retail  | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Datacenter x64           | OEM     | May or may not exist. If you have this, please give the key and/or ISO!           |                                                                                                         |
> 
> | Windows Server 2003 R2               | Datacenter x64           | Volume  | `B76XF-FCR39-XBGFF-YMBFH-G6PMT`                                                   | Duplicate of Standard x64, still works.                                                                 |
> 
> 
> 
> Just gotta deal with the weird one-off's like Home Server and stuff.

I must add that xp_activate32.exe will activate **all pre-reset** Longhorn builds in addition to post reset **build 5048** *in addition to* XP **and** Server 2003.

**Comment by ward201185** [comment]
Windows XP Professional SP0 Retail doesnt seem to accept the confirmation ids provided by xp_activate32.exe while Windows XP Home Edition SP0 Retail does.
later service pack versions of windows xp home and pro work just fine with xp_activate32.exe.
just adding this here if none of you knew.
xp_activate32.exe may need to be revised to accommodate all versions of the NT 5.X family.

**Comment by Neo-Desktop** [comment]
> Windows XP Professional SP0 Retail doesnt seem to accept the confirmation ids provided by xp_activate32.exe while Windows XP Home Edition SP0 Retail does.
> later service pack versions of windows xp home and pro work just fine with xp_activate32.exe.
> just adding this here if none of you knew.
> xp_activate32.exe may need to be revised to accommodate all versions of the NT 5.X family.


Just to clarify SP0 meaning RTM?

**Comment by ward201185** [comment]
> > Windows XP Professional SP0 Retail doesnt seem to accept the confirmation ids provided by xp_activate32.exe while Windows XP Home Edition SP0 Retail does.
> > later service pack versions of windows xp home and pro work just fine with xp_activate32.exe.
> > just adding this here if none of you knew.
> > xp_activate32.exe may need to be revised to accommodate all versions of the NT 5.X family.
> 
> Just to clarify SP0 meaning RTM?

yes, sorry for the confusion.

**Comment by Skiedeagle** [comment]
> Windows XP Professional RTM Retail doesnt seem to accept the confirmation ids provided by xp_activate32.exe while Windows XP Home Edition RTM Retail does. later service pack versions of windows xp home and pro work just fine with xp_activate32.exe. just adding this here if none of you knew. xp_activate32.exe may need to be revised to accommodate all non-beta versions of Windows XP and Server 2003.

Having this exact same issue but this time it's with an upgrade copy of XP Home using Service Pack 1. Maybe for some reason the application doesn't work with anything older than Service Pack 2 or 3 perhaps?

**Comment by diamondggg** [comment]
> > Windows XP Professional RTM Retail doesnt seem to accept the confirmation ids provided by xp_activate32.exe while Windows XP Home Edition RTM Retail does. later service pack versions of windows xp home and pro work just fine with xp_activate32.exe. just adding this here if none of you knew. xp_activate32.exe may need to be revised to accommodate all non-beta versions of Windows XP and Server 2003.
> 
> Having this exact same issue but this time it's with an upgrade copy of XP Home using Service Pack 1. Maybe for some reason the application doesn't work with anything older than Service Pack 2 or 3 perhaps?

You might be better to report this in the MDL topic to the actual author, rather than here ~~where people have trouble even figuring out how discrete logarithms of P and -P are related~~. Preferably with exact reproduction steps (you know, there are many variants of any version for different languages).

**Comment by Neo-Desktop** [comment]
> 
> You might be better to report this in the MDL topic to the actual author, rather than here ~~where people have trouble even figuring out how discrete logarithms of P and -P are related~~. Preferably with exact reproduction steps (you know, there are many variants of any version for different languages).

You know @diamondggg, I was under the impression that you were the actual author of the activation algorithm. Unfortunately it sounds like that's not the case

**Comment by ward201185** [comment]
> > Windows XP Professional RTM Retail doesnt seem to accept the confirmation ids provided by xp_activate32.exe while Windows XP Home Edition RTM Retail does. later service pack versions of windows xp home and pro work just fine with xp_activate32.exe. just adding this here if none of you knew. xp_activate32.exe may need to be revised to accommodate all non-beta versions of Windows XP and Server 2003.
> 
> Having this exact same issue but this time it's with an upgrade copy of XP Home using Service Pack 1. Maybe for some reason the application doesn't work with anything older than Service Pack 2 or 3 perhaps?

im not sure if there really are any upgrade editions of xp but maybe you would have a better chance of activation working if you had changed your product key to a non-upgrade key because xp_activate32.exe did definitely work for me on windows xp home edition sp1 retail.

**Comment by thepwrtank18** [comment]
Does anyone have a list of Upgrade-only ISO's I can use for testing?

**Comment by ward201185** [comment]
> > > Windows XP Professional RTM Retail doesnt seem to accept the confirmation ids provided by xp_activate32.exe while Windows XP Home Edition RTM Retail does. later service pack versions of windows xp home and pro work just fine with xp_activate32.exe. just adding this here if none of you knew. xp_activate32.exe may need to be revised to accommodate all non-beta versions of Windows XP and Server 2003.
> > 
> > 
> > Having this exact same issue but this time it's with an upgrade copy of XP Home using Service Pack 1. Maybe for some reason the application doesn't work with anything older than Service Pack 2 or 3 perhaps?
> 
> im not sure if there really are any upgrade editions of xp but maybe you would have a better chance of activating successfully if you had changed your product key to a non-upgrade key because xp_activate32.exe did definitely work for me on windows xp home edition sp1 retail.
> 
> maybe your specific product key is the issue.

i just tried a key that is supposedly an upgrade key.
it still activated just fine with xp_activate32.exe even in sp1.
heres the key: C6QQT-DYQ7H-HG39T-H47FX-Q3JD2
maybe someone could try verifying it by calculating the upgrade bit in that key.

**Comment by Skiedeagle** [comment]
Well it's a genuine copy, box, manuals and everything, Even has the upgrade label on the holographic disc and box. I wasn't really thinking much when I was doing testing for it. The key from the box is: HB49T-J3KGF-B2PM7-RM29P-CXG6F works fine with the activation hotline but once it spits out the confirmation from the xp_activate32 app and I entered it, give me an error.

**Comment by thepwrtank18** [comment]
Can you copy the disc to an ISO and send it here?

**Comment by Neo-Desktop** [comment]
[imgburn](https://www.imgburn.com/) is also prefered - please upload to [archive.org](https://archive.org/)

**Comment by ward201185** [comment]
> Well it's a genuine copy, box, manuals and everything, Even has the upgrade label on the holographic disc and box. I wasn't really thinking much when I was doing testing for it. The key from the box is: HB49T-J3KGF-B2PM7-RM29P-CXG6F works fine with the activation hotline but once it spits out the confirmation from the xp_activate32 app and I entered it, give me an error.

i just put in your product key and tried activating via xp_activate32.exe and it worked just fine.
sounds like your installation may be different.

**Comment by Skiedeagle** [comment]
Weird, guess it's the upgrade disc I have then, I'm in the process of uploading it now, just waiting through the long upload progress rn. Just kinda weird that it would be the upgrade disc that's affecting the product activation or the supposed service pack on the disc.

**Comment by ward201185** [comment]
dont think its the service pack (maybe it is), think it has something to do with the fact its an upgrade edition.
your system files might be different (pidgen.dll, licdll.dll, and etc)

**Comment by Skiedeagle** [comment]
interesting, nevertheless here's the upgrade copy I imaged https://archive.org/details/xrmhccp-en_202306

**Comment by ward201185** [comment]
https://www.angelfire.com/comics/observer/wxpvol.htm
so it seems like upgrade edition exists for xp rtm, sp1, and sp2 but maybe not for sp3.

**Comment by ward201185** [comment]
also found out that upgrade versions of windows xp already exist on archive.org, sorry if that wasted some of your time.

**Comment by ward201185** [comment]
> Weird, guess it's the upgrade disc I have then, I'm in the process of uploading it now, just waiting through the long upload progress rn. Just kinda weird that it would be the upgrade disc that's affecting the product activation or the supposed service pack on the disc.

i used your xp iso and installed via upgrade and the clean install approach.
xp_activate32.exe still activates just fine.
are you sure you inputted the installation id correctly?

**Comment by Neo-Desktop** [comment]
here's some archive.org links I've found

https://archive.org/details/wxpccp-en
https://archive.org/details/xrmpccp-en
https://archive.org/details/winp-professional-sp1-upgrade (X1APCCP_EN)
https://archive.org/details/Windows-XP-Professional-ISO-Version-2002-SP2-OEM-Disc-Image

https://archive.org/details/wxhccp-en
https://archive.org/details/xrmhccp-en
https://archive.org/details/x-1-ahccp-en
https://archive.org/details/WinXPHomeSP2Upgrade

**Comment by Skiedeagle** [comment]
One time I think I tried the get and store from the app and manually typing it in Virtual PC 2004, the id was exactly as it was typed out. I tried again, gave me a different one and that didn't work either. I'm trying it again on virtualbox now but hopefully it wasn't any minor mistakes on my end.

**Comment by thepwrtank18** [comment]
I'm testing Windows 95, 98, 98 SE, NT 4.0, 2000, and ME to see if it works on Home and Professional, the only editions of XP that had Upgrade discs.

**Comment by Skiedeagle** [comment]
huh, putting it in virtualbox seemed to work, so sorry if I may have wasted all your time. Weird it was the right version, I think the program didn't seem to work for me in virtual pc for some reason

**Comment by Skiedeagle** [comment]
don't know if it had something to do with the drag and drop method or what, gonna try it again and see if it was a mistake

**Comment by ward201185** [comment]
maybe vpc 2004 had some form of oem bios embedded into it and windows xp acted like an oem installation so it blocked any form of telephone/online activation?

**Comment by Skiedeagle** [comment]
Nope, didn't seem to be the case. Tried it again on VPC 2004, and for some reason it worked. Maybe because I did the get and store commands first, but I couldn't remember if there were any mistakes when I entered the ID from VPC in. Maybe it was my fault and I was just wasting my time by doing it, but can't keep wondering to myself how it was able to work that time. Well I appreciate the effort nevertheless

**Comment by ward201185** [comment]
did you activate on the same vm or did you install on another vm?

**Comment by Skiedeagle** [comment]
I created a new vm for the install after the first one failed on me. I think I was installing the first one directly from the disc if I recall correctly

**Comment by Neo-Desktop** [comment]
just fyi: we are migrating away from using `xp_activate32.exe` and instead focusing on `xpkey -i` (as seen in the readme)

**Comment by Skiedeagle** [comment]
Good to see the hard work pay itself off. I gotten the error a few days ago and didn't think much of it. Hope to see more improvement in old Microsoft activations in the future
